Library/src/Multiplayer/SpaceEntity.cpp

```diff
@@ -1191,6 +1247,52 @@ ComponentBase* SpaceEntity::FindFirstComponentOfType(ComponentType Type, bool Se
 	return LocatedComponent;
 }
 
+void SpaceEntity::AddChildEntitiy(SpaceEntity* ChildEntity)
+{
+	ChildEntities.Append(ChildEntity);
+}
+
+bool SpaceEntity::ResolveEntityHierarchy()
+{
+	if (ShouldUpdateParent == false)
+	{
+		return false;
+	}
+
+	// Entity has been re-parented
+	if (ParentId.HasValue())
+	{
+		// Entity was previously parented to another object, so cleanup
+		if (Parent != nullptr)
+		{
+			Parent->ChildEntities.RemoveItem(this);
+		}
+
+		// Set our new parent
+		Parent = GetSpaceEntitySystem()->FindSpaceEntityById(*ParentId);
+
+		if (Parent != nullptr)
+		{
+			Parent->ChildEntities.Append(this);
+		}
+		else
+		{
+			CSP_LOG_ERROR_MSG("SpaceEntity unable to find parent");
```

Comment:
Worth including the parent ID in the message to aid debugging?

Reply:
@MAG-SamBirley @MAG-AW. The GetObject call will make it more robust. The only downside is that this is another async call which will add additional time/complexity.

Is this something we feel should be done in this PR?

Reply:
So should we just go with the parent ID for this PR?